### PR TITLE
Bump to `darwinia-v0.13.0`

### PR DESCRIPTION
**Pull request by AsceticBear**
The pr mainly brings the latest `darwinia-messages-substrate` into the pangolin and pangoro network. 

Includes:
1. Companion for https://github.com/darwinia-network/darwinia-messages-substrate/pull/206
2. Companions for fee market latest bug fixes.

TODO:
1. migration for `pallet-messages` and `pallet-grandpa`, traced by #1465

### CODE REVIEW

File: node/runtime/pangolin/src/pallets/bridge_parachains.rs [hunk 1]
```diff
@@ -4,13 +4,28 @@ pub use pallet_bridge_parachains::{
...
+	// TODO: Is it okay to use Everything here
+	pub GetTenFirstParachains: Vec<ParaId> = (0..10).map(ParaId).collect();
+}
+
 impl Config<WithRococoParachainInstance> for Runtime {
 	type BridgesGrandpaPalletInstance = WithRococoGrandpa;
+	type Event = Event;
 	type HeadsToKeep = RococoHeadersToKeep;
+	type ParasPalletName = ParasPalletName;
+	type TrackedParachains = frame_support::traits::Everything;
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/pallets/bridge_parachains.rs:21]: @jiguantong Please help check for this, Is `Everything` okay in our parachain scenarios?

File: node/runtime/pangolin/src/pallets/bridge_parachains.rs [hunk 2]
```diff
@@ -4,13 +4,28 @@ pub use pallet_bridge_parachains::{
 
 // --- darwinia-network ---
 use crate::*;
+use bp_polkadot_core::parachains::{ParaId, PARAS_PALLET_NAME};
 use pallet_bridge_parachains::Config;
 
+frame_support::parameter_types! {
+	pub const ParasPalletName: &'static str = PARAS_PALLET_NAME;
+	// TODO: Is it okay to use Everything here
+	pub GetTenFirstParachains: Vec<ParaId> = (0..10).map(ParaId).collect();
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/pallets/bridge_parachains.rs:13]: Or a filter like this.

File: node/runtime/pangolin/src/pallets/bridge_parachains.rs (hunk 1 above)
Comment (jiguantong) [reply node/runtime/pangolin/src/pallets/bridge_parachains.rs:21]: @fewensa It's used [here](https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/parachains/src/lib.rs#L273). And related upstream pr is https://github.com/paritytech/parity-bridges-common/pull/1432.
Should we track `Everything` here or a specific parachain? like 2105(pangolin parachain)

Comment (fewensa) [reply node/runtime/pangolin/src/pallets/bridge_parachains.rs:21]: Nope, it's not used by the bridger.

Comment (jiguantong) [reply node/runtime/pangolin/src/pallets/bridge_parachains.rs:21]: @AsceticBear Leave it as `Everything` for now

File: node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs [hunk 3]
```diff
@@ -147,23 +148,23 @@ impl ThisChainWithMessages for Pangolin {
 #[derive(Clone, Copy, RuntimeDebug)]
 pub struct PangolinParachain;
 impl ChainWithMessages for PangolinParachain {
-	type AccountId = bp_pangolin_parachain::AccountId;
```
Comment (AurevoirXavier) [inline node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs:150]: Why?

File: node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs [hunk 4]
```diff
@@ -124,11 +125,11 @@ impl MessageBridge for WithPangolinParachainMessageBridge {
 #[derive(Clone, Copy, RuntimeDebug)]
 pub struct Pangolin;
 impl ChainWithMessages for Pangolin {
-	type AccountId = bp_pangolin::AccountId;
```
Comment (AurevoirXavier) [inline node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs:127]: Why?

**Comment by AurevoirXavier** [comment]
I prefer to keep all the `bp*::`. We can know where those primitives come from.

**Comment by AsceticBear** [comment]
> I prefer to keep all the `bp*::`. We can know where those primitives come from.

image

As the screenshot shows, most of the primitives used in this repo now come from `bp-darwinia-core` and these primitives are not overlapped with `bp-darwinia-core`'s. So I think it is acceptable to omit the `bp-*` prefix.

**Comment by HackFisher** [comment]
> I prefer to keep all the bp*::. We can know where those primitives come from.

I think keep this prefix here will make it more readable and accurate, do not need to search around or tip for finding source of primitive.

**Comment by AurevoirXavier** [comment]
> As the screenshot shows, most of the primitives used in this repo now come from `bp-darwinia-core` and these primitives are not overlapped with `bp-darwinia-core`'s. So I think it is acceptable to omit the `bp-*` prefix.

I suggest `pub use bp_darwina_core as bp_pangolin_core`, ...

**Comment by AsceticBear** [comment]
> > I prefer to keep all the bp*::. We can know where those primitives come from.
> 
> I think keep this prefix here will make it more readable and unambiguous, do not need to search around or tip for finding source of primitive.

Fine. I'll push another commit to fix this. Please review the other changes first.

File: node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs (hunk 3 above)
Comment (AsceticBear) [reply node/runtime/pangolin/src/bridges_message/pangolin_parachain.rs]: Fixed. Please review it again. @AurevoirXavier @HackFisher

File: node/runtime/pangolin/src/bridges_message/pangolin_parachain_alpha.rs [hunk 5]
```diff
@@ -156,14 +159,15 @@ impl ChainWithMessages for PangolinParachainAlpha {
 }
 impl BridgedChainWithMessages for PangolinParachainAlpha {
 	fn maximal_extrinsic_size() -> u32 {
-		bp_pangolin_parachain::PangolinParachain::max_extrinsic_size()
+		PangolinParaChain::max_extrinsic_size()
```
Comment (AurevoirXavier) [inline node/runtime/pangolin/src/bridges_message/pangolin_parachain_alpha.rs:162]: Keep the same style as others?

Or change all `darwinia_common_runtime::Chain::max_extrinsic_siz`e to this.

File: node/runtime/common/src/lib.rs [hunk 6]
```diff
@@ -27,9 +27,24 @@ pub use gov_origin::*;
...
-pub use darwinia_balances::{Instance1 as RingInstance, Instance2 as KtonInstance};
+/// Primitives of the Pangolin chain.
+pub use bp_darwinia_core as bp_pangolin;
+/// Primitives of the Pangolin-parachain and Pangolin-parachain-alpha chain.
+pub use bp_darwinia_core as bp_pangolin_parachain;
+/// Primitives of the Pangoro chain.
+pub use bp_darwinia_core as bp_pangoro;
+/// Re-export DarwiniaLike as different chain type.
+pub use bp_darwinia_core::{
+	DarwiniaLike as Pangolin, DarwiniaLike as Pangoro, DarwiniaLike as PangolinParaChain,
```
Comment (AurevoirXavier) [inline node/runtime/common/src/lib.rs:41]: Typo `ParaChain`

Comment (AsceticBear) [reply node/runtime/common/src/lib.rs:41]: Do you mean renaming `PangolinParaChain` to `ParaChain`?

Comment (AurevoirXavier) [reply node/runtime/common/src/lib.rs:41]: Parachain

**Pull request merged**